dips/dip-158.md

```diff
@@ -0,0 +1,805 @@
+
+## P2M Payment Types
+There are two types of P2M payments described in this document: 
+1. **Charge**. Charge payment stipulates an immediate capture of funds. This means that a successful off-chain negotiation for such payment will result in funds being captured 
+2. **Auth / Capture**. Auth / Capture payment, as the name suggests, means that the payment is divided to two steps: Authorization and Capture. Successful Authorization implies that the funds are locked so they can later be captured
+---
+# Glossary
+For the sake of simplicity, we use the following terms:
+* "Sender", "Payer", "Wallet" or "customer VASP" - The customer's wallet. It is the VASP representing the customer. This is the VASP that will send the funds and eventually will submit the on-chain transaction to the blockchain
+* "Receiver", "Receiving VASP" or "Merchant VASP" - The VASP representing the merchant which receives the funds (transaction)
+* "Customer" - The person who wishes to pay the merchant for goods or services
+* "Reference ID", "Common Payment Identifier", "Payment Identifier" - A unique identifier of the payment. It must be a UUID according to RFC4122 with "-"'s included. See also [Reference ID in DIP-1](https://github.com/diem/dip/blob/main/dips/dip-1.mdx#terminology)
+---
+# Prerequisite: Sharing Payment Identifier and Receiver Address
+A prerequisite for the off-chain negotiations is that both parties (sending and receiving VASPs) have shared a common payment identifier and the receiver address. This common payment idenbtifier is required to support a successive off-chain protocol between the VASPs. 
+This DIP specifies a couple of examples for exchanging identifiers between VASPs, however, there could be others. The protocol is agnostic to the method of exchanging the common payment identifier. 
+
+See [Appendix A](#appendix-a---prerequisite-sharing-common-payment-identifier-and-address)
+
+---
+# P2M Off-Chain Protocol Specification
+This section specifies P2M payment types, required structures, payloads and expected sequence of calls based on this protocol.
+
+Content:
+* [High Level Flow and Payment Types](#high-level-flow-and-payment-types)
+* [Sender / Receiver Interface](#sender--receiver-interface)
+* [Request / Response Payload](#request--response-payload)
+* [Payment Sequences](#payment-sequences)
+* [New Structures](#new-structures)
+* [Appendix A: Sharing Common Payment Identifier and Address](#appendix-a---prerequisite-sharing-common-payment-identifier-and-address)
+---
+## High-Level Flow and Payment Types
+---
+The high-level flow of an eCommerce P2M payment (regardless of the type) is:
+1. On the merchant's checkout page the customer chooses to pay with Diem and selects the preferred Diem wallet to use
+2. The merchant's VASP (the receiver) generates a payment request with all details needed to identify this payment later in the process and makes it available to the customer's Diem wallet (the customer's VASP or sender)
```

Comment:
I think we're leaving out the detail where the merchant transmits the payment request to their VASP.

Reply:
Added